components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -2509,6 +2508,147 @@ public void updateSubscriptionStatusAndTier(int subscriptionId, String status) t
         }
     }
 
+    public List<Integer> getSubscribersForOrganizationId(String organization) throws APIManagementException {
+        List<Integer> subscriberIdList = new ArrayList<>();
+        String query = SQLConstants.GET_SUBSCRIBERS_FOR_ORG_ID;
+
+        try (Connection connection = APIMgtDBUtil.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(query);
+        ){
+            preparedStatement.setString(1, organization);
+            try (ResultSet rs = preparedStatement.executeQuery();){
+                while (rs.next()) {
+                    subscriberIdList.add(rs.getInt("SUBSCRIBER_ID"));
+                }
+            }
+        } catch (SQLException e) {
+            handleException("Error while retrieving subscribers for organization " +organization, e);
+        }
+
+        return subscriberIdList;
+    }
+
+    public List<String> getMappedOrganizationListForSubscriber(int subscriberId) throws APIManagementException {
+        List<String> organizationList = new ArrayList<>();
+        String query = SQLConstants.GET_MAPPED_ORGANIZATIONS_FOR_SUBSCRIBER_ID;
+
+        try (Connection connection = APIMgtDBUtil.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(query);
+        ){
+            preparedStatement.setInt(1, subscriberId);
+            try (ResultSet rs = preparedStatement.executeQuery();){
+                while (rs.next()) {
+                    organizationList.add(rs.getString("ORGANIZATION"));
+                }
+            }
+        } catch (SQLException e) {
+            handleException("Error while retrieving organizations for subscriber id " +subscriberId, e);
+        }
+        return organizationList;
+    }
+
+    public void removeSubscriber(int subscriberId) throws APIManagementException {
+        Connection connection = null;
+        PreparedStatement preparedStatement = null;
+        String query = SQLConstants.DELETE_SUBSCRIBER;
+        try {
+            connection = APIMgtDBUtil.getConnection();
+            preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setInt(1, subscriberId);
+            preparedStatement.executeUpdate();
+
+            connection.close();
+            preparedStatement.close();
+        } catch (SQLException e) {
+            handleException("Error while removing subscriber with subscriber id " +subscriberId, e);
+        }
```

Comment:
connection not closed for the exception case. either use finally block or use try-with-resource method. see http://www.mastertheboss.com/jbossas/jboss-datasource/using-try-with-resources-to-close-database-connections/